src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -3090,9 +3090,89 @@ public void Test_Issue6423()
         }
 
         /// <summary>
-        /// https://github.com/Azure/bicep/issues/3169
+        /// https://github.com/Azure/bicep/issues/3356
+        /// </summary>
+        [TestMethod]
+        public void Test_Issue_3356()
+        {
+            var result = CompilationHelper.Compile(@"
+resource foo 'Microsoft.Storage/storageAccounts@2021-09-00' = {
```

Comment:
[nit] What about having a `disable` directive so that we can just say `result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics()` - I feel like that states the intention of this test a bit more clearly.

```suggestion
            var result = CompilationHelper.Compile(@"
#disable-next-line BCP081
resource foo 'Microsoft.Storage/storageAccounts@2021-09-00' = {
```

I would also recommend either naming the tests or adding comments to them to explain the purpose (validating we have default type definitions that accept well-formed inputs / warn on bad inputs)